File: 🟢 Easy/263 - Ugly Number/_263UglyNumber.java
class Solution {
    public boolean isUgly(int n) {
        if (n <= 0) {
            return false;
        }

        int[] primeFactors = { 2, 3, 5 };
        for (int prime : primeFactors) {
            while (n % prime == 0) {
                n /= prime;
            }
        }

        // if 1 occurs then it is "UGLY NUMBER".
        return n == 1;
    }
}

File: 🟢 Easy/263 - Ugly Number/_263UglyNumber.c
bool isUgly(int n) {
    if (n <= 0) {
        return false;
    }

    int primeFactors[] = {2, 3, 5};
    for (int i = 0; i < 3; i++) {
        while (n % primeFactors[i] == 0) {
            n /= primeFactors[i];
        }
    }

    // if 1 occurs then it is "UGLY NUMBER".
    return n == 1;
}